src/compose/ComposeBox.js

```diff
@@ -423,6 +448,22 @@ class ComposeBox extends PureComponent<Props, State> {
   }
 }
 
+const getPmrecipientIsDeactivated = (state: GlobalState, narrow: Narrow): boolean => {
+  if (!isPrivateNarrow(narrow)) {
+    return false;
+  }
+  const recipientEmail = narrow[0].operand;
+  return !getUserIsActive(state, recipientEmail);
+};
+
+const getGroupPmrecipientIsDeactivated = (state: GlobalState, narrow: Narrow): boolean => {
+  if (!isGroupNarrow(narrow)) {
+    return false;
+  }
+  const recipientEmails: string[] = narrow[0].operand.split(',');
```

Comment:
I considered using the selector `getRecipientsInGroupNarrow` here, but it would create un-necessary User objects when we only require emails.

Reply:
In `getRecipientsInGroupNarrow`, it looks like various containers for the `User` objects (arrays, and a `Map`) do get re-created (though not the `User` objects themselves, I think). I guess the performance cost is linear with the number of (total?) users, and we do avoid that concern by inspecting the operand here. Some caching is done in `getRecipientsInGroupNarrow`; that may help performance if we were to use that.

One thing about doing `narrow[0].operand.split(',');` here is that it's dealing directly with the raw data, so this code would need to be revisited if the shape of that data changed. Not the end of the world, but I like to reach first for reusable functions that do that work, if possible. I wonder if `emailsOfGroupNarrow` might be appropriate, if `getRecipientsInGroupNarrow` has too high a performance cost? @gnprice, maybe you have some thoughts? 🙂

Reply:
@chrisbobbe this is moot now, since `getUserIsActive` now accepts user IDs only.